lib/ukfallocbuddy/tests/test_ukfallocbuddy.c

```diff
@@ -0,0 +1,217 @@
+#include <uk/test.h>
+#include <uk/fallocbuddy.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+struct test_ukfallocbuddy {
+
+	struct uk_falloc *fa;
+	__paddr_t start;
+	__sz len;
+	unsigned long pages;
+	__sz fa_meta_size;
+	void *start_zone;
+	void *fa_meta;
+};
+
+
+static struct uk_falloc *init_new_fb_alloc(void *fa_meta, void *start_zone,
+	unsigned long len, __paddr_t *new_start, __sz fa_meta_size,
+	unsigned long *pages)
+{
+	__vaddr_t dm_off;
+	__paddr_t start = (__paddr_t)start_zone;
+	struct uk_falloc *fa;
+
+	UK_ASSERT(fa = malloc(uk_fallocbuddy_size()));
+
+	UK_ASSERT(!uk_fallocbuddy_init(fa));
+	start = PAGE_ALIGN_UP(start + fa_meta_size);
+	*pages = (len - fa_meta_size) >> PAGE_SHIFT;
+	dm_off = start;
+	UK_ASSERT(!fa->addmem(fa, fa_meta, start, *pages, dm_off));
+	*new_start = start;
+	return fa;
+}
+
+static struct test_ukfallocbuddy *initialize_parameters(__sz len)
+{
+	struct test_ukfallocbuddy *test_attributes =
+(struct test_ukfallocbuddy *) malloc(sizeof(struct test_ukfallocbuddy));
+
+	UK_ASSERT(test_attributes);
+
+	test_attributes->len = len;
+	test_attributes->pages = len >> PAGE_SHIFT;
+	test_attributes->fa_meta_size =
+		uk_fallocbuddy_metadata_size(test_attributes->pages);
+	test_attributes->start_zone = malloc(len);
+	UK_ASSERT(test_attributes->start_zone);
+	test_attributes->fa_meta = malloc(test_attributes->fa_meta_size);
+	UK_ASSERT(test_attributes->fa_meta_size);
+	test_attributes->fa = init_new_fb_alloc(test_attributes->fa_meta,
+						test_attributes->start_zone,
+						test_attributes->len,
+						&(test_attributes->start),
+						test_attributes->fa_meta_size,
+						&(test_attributes->pages));
+	UK_ASSERT(test_attributes->fa);
+
+	return test_attributes;
+
+}
+
+static void free_memory(struct test_ukfallocbuddy *test_attr)
```

Comment:
It makes sense to match the function's name with the allocation function.
```suggestion
static void free_allocator(struct test_ukfallocbuddy *test_attr)
```

Reply:
Yes. It makes sense. Went with your suggestion.